activerecord/lib/active_record/model_schema.rb

```diff
@@ -484,6 +484,9 @@ def load_schema
         end
 
         def load_schema!
+          unless table_exists?
```

Comment:
This will be true in two scenarios: when the model is abstract and no table is configured, or when a table is configured but it doesn't exist in the database. In the latter case I think the existing error works better, since it's clearer that the application expected the table to exist.

I think we should only raise this new error in the former case:

```suggestion
          unless table_name
```

This would conveniently also fix the railties tests that are failing:
https://buildkite.com/rails/rails/builds/61157#fdf33e0b-7168-4e4a-bb9c-ed39c4f40766/1019-1029

Reply:
great catch thanks!